backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java

```diff
@@ -108,4 +113,18 @@ public ProposalRevisionEmail requestChanges(
     private boolean hasFeedback(ProposalRevisionDto proposalRevisionDto) {
         return proposalRevisionDto.getFeedback() != null;
     }
+
+    public ChangeStatusToPublishedResult changeStatusToPublished(String idProposal) {
+        JpaProposal proposal = jpaProposalRepository.findByNaturalId(idProposal).orElseThrow(EntityNotFoundException::new);
+        Integer status = proposal.getStatus().getId();
+        if (status.equals(FINISHED.getId()) || status.equals(INADEQUATE.getId()) || status.equals(CANCELLED.getId())) {
+            throw new IllegalStateException();
+        }
```

Comment:
There are 3 cases to test, look for them. (SonarCloud issue)

Don't forget the 4XX tests.